src/middlewared/middlewared/plugins/service_monitor.py

```diff
@@ -67,50 +65,85 @@ def isEnabled(self, service):
         # start the service.
         #
         if service in ('activedirectory', 'ldap', 'nis'):
-            ds = self.middleware.call_sync('datastore.query', 'directoryservice.%s' % service)[0]
             if service == 'activedirectory':
                 service = 'ad'
-            enabled = ds["%s_enable" % service]
-
-        else:
-            services = self.middleware.call_sync('datastore.query', 'services.services')
-            for s in services:
-                if s['srv_service'] == 'cifs':
-                    enabled = s['srv_enable']
-                # What about other services?
```

Comment:
What if the services settings are changed while the monitoring is running? Would using the cached values be a problem somehow?

Reply:
Right now the only service we monitor is AD, and we restart service monitor whenever we change AD settings. We might need to re-evaluate when / if we decide to start using this to monitor other services.